components/constellation/frame.rs

```diff
@@ -22,15 +24,21 @@ pub struct Frame {
     /// The frame id.
     pub id: FrameId,
 
-    /// The timestamp for the current session history entry
+    /// The size of the frame.
+    pub size: Option<TypedSize2D<f32, CSSPixel>>,
+
+    /// The timestamp for the current session history entry.
     pub instant: Instant,
 
-    /// The pipeline for the current session history entry
+    /// The pipeline for the current session history entry.
     pub pipeline_id: PipelineId,
 
-    /// The load data for the current session history entry
+    /// The load data for the current session history entry.
     pub load_data: LoadData,
 
+    /// The type of document for the current session history entry.
+    pub doc_type: DocumentType,
```

Comment:
On thinking about this, why does the constellation need to track this? Can we not just track this in script?